imod_coupler/drivers/ribametamod/ribametamod.py

```diff
@@ -305,27 +305,37 @@ def couple(self) -> None:
+                    first_id = id(self.ribasim_user_demand)
                     modribmsw_arrays["rib_sprinkling_demand"] = self.ribasim_user_demand
-                    n_users = np.shape(self.mapping.msw2rib["sw_sprinkling"])[0]
-                    n_priorities = np.size(self.ribasim_user_demand) // n_users
-                    self.ribasim_user_demand.resize(n_users, n_priorities)
-                    self.ribasim_userprio = self.ribasim_user_demand.copy().reshape(
-                        n_users, n_priorities
-                    )
-                    self.ribasim_userprio[self.ribasim_userprio > 1] = 1.0
-                    # number of priorities non-zero elements for each NON-MASKED user, i.e. metaswap sprinkling
-                    nprio_per_target = (
-                        1 - self.mapping.msw2rib["sw_sprinkling_mask"]
-                    ) * self.ribasim_userprio.sum(axis=1)
-
-                    # gather 1-based indices of users with more than one entry into a list
-                    too_many = np.where(nprio_per_target > 1)[0] + 1
-
-                    # if this list is not empty ...
-                    if np.size(too_many) > 0:
+                    n_users = self.subgrid_level.size
+                    n_priorities = self.ribasim_user_demand.size // n_users
+                    self.ribasim_user_demand.resize(n_priorities, n_users)
+                    if first_id != id(self.ribasim_user_demand):
                         raise ValueError(
-                            f"More than one priority set for basins {np.array2string(too_many)}"
+                            "reshaping user demand array resulted in a copy of array"
                         )
+                    self.coupled_user_indices = np.flatnonzero(
+                        self.mapping.msw2rib["sw_sprinkling_mask"] == 0
+                    )
+                    self.coupled_priority_indices = np.flatnonzero(
+                        self.ribasim_user_demand[:, self.coupled_user_indices] != 0.0
+                    )
+
+                    # self.ribasim_userprio = self.ribasim_user_demand.copy()
+                    # self.ribasim_userprio[self.ribasim_userprio > 1] = 1.0
+                    # # number of priorities non-zero elements for each NON-MASKED user, i.e. metaswap sprinkling
+                    # nprio_per_target = (
+                    #     1 - self.mapping.msw2rib["sw_sprinkling_mask"]
+                    # ) * self.ribasim_userprio.sum(axis=1)
+                    #
+                    # # gather 1-based indices of users with more than one entry into a list
+                    # too_many = np.where(nprio_per_target > 1)[0] + 1
+                    #
+                    # # if this list is not empty ...
+                    # if np.size(too_many) > 0:
+                    #     raise ValueError(
+                    #         f"More than one priority set for basins {np.array2string(too_many)}"
+                    #     )
 
```

Comment:
For keeping track of the violations of the constraint of one priority per sprinkling user demand maybe something like:
                    # * True-False array on non-zero value from self.ribasim_user_demand
                    # * Count True (=sum) by column, axis=0
                    # * Keep indices with count beyond one in a list
                    # * Intersect that list with coupled_user_indices
                    too_many = np.intersect1d(
                       np.where(np.sum(self.ribasim_user_demand!=0,axis=0)>1),
                       self.coupled_user_indices
                    )
# throw a value error with the one-based list of user indices violating the rule
                    if np.size(too_many) > 0:
                        raise ValueError(
                            f"More than one priority set for basins {np.array2string(too_many)+1}"
                        )